sonarts-core/tests/rules/preferTypeGuardRule/preferTypeGuardRule.lint.ts

```diff
@@ -0,0 +1,68 @@
+  function isFish(animal: Animal) {
+//^^^^^^^^ {{Change this boolean return type into the type predicate "animal is Fish".}}
+  return (animal as Fish).swim !== undefined;
+}
+
+function isFish(animal: Animal): animal is Fish {
+  return (animal as Fish).swim !== undefined;
+}
+
+  function isFish(animal: Animal) {
```

Comment:
what's the difference with the first case?

Reply:
Here I'm testing `(animal as Fish).swim != undefined;` while in the first case is `return (animal as Fish).swim !== undefined;` (i.e. strict inequality)

Reply:
indeed:) that's why it's nicer to have descriptive test cases (with comment above or meaningful function name)

Reply:
Point taken, I will add a comment!